contracts/DCAHubSwapper/DCAHubSwapperSwapHandler.sol

```diff
@@ -36,12 +36,26 @@ abstract contract DCAHubSwapperSwapHandler is DeadlineValidation, SwapAdapter, I
   using SafeERC20 for IERC20;
   using Address for address;
 
+  bytes32 public constant ADMIN_ROLE = keccak256('ADMIN_ROLE');
+  bytes32 public constant SWAP_EXECUTION_ROLE = keccak256('SWAP_EXECUTION_ROLE');
+
   /// @notice Represents the lack of an executor. We are not using the zero address so that it's cheaper to modify
   address internal constant _NO_EXECUTOR = 0x000000000000000000000000000000000000dEaD;
   /// @notice The caller who initiated a swap execution
   address internal _swapExecutor = _NO_EXECUTOR;
 
-  constructor(address _swapperRegistry) SwapAdapter(_swapperRegistry) {}
+  constructor(
+    address _swapperRegistry,
+    address _admin,
+    address[] memory _initialSwapExecutors
+  ) SwapAdapter(_swapperRegistry) {
+    if (_admin == address(0)) revert ZeroAddress();
+    _setupRole(ADMIN_ROLE, _admin);
```

Comment:
As said off-github, remember adding ADMIN_ROLE as its admin :)

Reply:
Added in #118